rapidhash.h

```diff
@@ -226,20 +249,112 @@
   *
   *  Returns a 64-bit value containing all three bytes read.
   */
-  RAPIDHASH_INLINE_CONSTEXPR uint64_t rapid_readSmall(const uint8_t *p, size_t k) RAPIDHASH_NOEXCEPT { return (((uint64_t)p[0])<<56)|(((uint64_t)p[k>>1])<<32)|p[k-1];}
- /*
-  *  rapidhash main function.
-  *
-  *  @param key     Buffer to be hashed.
-  *  @param len     @key length, in bytes.
-  *  @param seed    64-bit seed used to alter the hash result predictably.
-  *  @param secret  Triplet of 64-bit secrets used to alter hash result predictably.
-  *
-  *  Returns a 64-bit hash.
-  */
+ RAPIDHASH_INLINE_CONSTEXPR uint64_t rapid_readSmall(const uint8_t *p, size_t k) RAPIDHASH_NOEXCEPT { return (((uint64_t)p[0])<<56)|(((uint64_t)p[k>>1])<<32)|p[k-1];}
+
+/**
+ * When using sequential seeds, premix them to avoid similarity between seeds.
+ *
+ * @ param seed 64-bit seed.
+ *
+ * @ return 64-bit premixed seed.
+ */
+RAPIDHASH_INLINE_CONSTEXPR uint64_t rapid_preseed(const uint64_t seed) RAPIDHASH_NOEXCEPT {
+  return rapid_mix(seed ^ rapid_secret[2], rapid_secret[1]);
+}

```

Comment:
This is the pre-seeding that has been removed from the hotpath, and is instead presented to the user as a separate function should they need it. A well chosen fixed, random, or default seed shouldn't need this and it can't always be optimised out -- for example is runtime random seeds.

Ups the chances of user error using seeds 0, 1, 2 but makes a 10% difference on small inputs. Are you happy with this from a user experience?